RecoLocalCalo/HcalRecAlgos/test/HcalRecHitReflagger.cc

```diff
@@ -201,6 +185,21 @@ HcalRecHitReflagger::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
    iSetup.get<HcalRecNumberingRecord>().get(topo_);
    topo = &*topo_;
 
+   //get channel quality conditions 
+   edm::ESHandle<HcalChannelQuality> p;
+   iSetup.get<HcalChannelQualityRcd>().get("withTopo",p);
+   const HcalChannelQuality& chanquality_(*p.product());
+   
+   std::vector<DetId> mydetids = chanquality_.getAllChannels();
+   for (std::vector<DetId>::const_iterator i = mydetids.begin();i!=mydetids.end();++i)

```

Comment:
this looks like a pretty expensive operation to be done every event.
Some ```ESWatcher``` should be added to update only when necessary